### PR TITLE
NIFI-3639 - Implemented HBase Get functionality for HBaseClientService

### PR DESCRIPTION
**Pull request by baolsen**
NIFI-3639 - Implemented HBase Get functionality for HBaseClientService
Code changes have been tested with current NiFi / maven build, which completed successfully.

- This is my first PR on Github, please let me know if there is anything I could improve on or if anything is incomplete.

### CODE REVIEW

**Comment by bbende** [comment]
Hi @baolsen , thanks for contributing! 

One question, is there any difference between the get method being added, and using the existing scan method with startRow and endRow being the same value?

There is a new FetchHBaseRow processor in master that hasn't been released yet, and its using scan to fetch a single row:

https://github.com/apache/nifi/blob/master/nifi-nar-bundles/nifi-hbase-bundle/nifi-hbase-processors/src/main/java/org/apache/nifi/hbase/FetchHBaseRow.java#L259-L262

**Comment by baolsen** [comment]
Hi @bbende, my pleasure!

Thanks for the quick response.

The intention was to use the Get for another processor I am writing which just needs single row lookups (didn't at first realise that FetchHBaseRow was also doing single row lookups). 

I had assumed that a Get would be more efficient than a Scan for fetching single rows.
 
However, upon further reading it seems that the HBase client API uses a Scan implementation for Gets as well. https://www.cloudera.com/documentation/enterprise/5-4-x/topics/admin_hbase_scanning.html

There are some stack overflow questions regarding Get performance being poorer than Scan, especially when using a key prefix in the scan as opposed to a full rowkey.
https://www.quora.com/What-is-the-difference-between-get-and-scan-in-HBase

It's a little unclear what scenarios cause this performance difference, or whether one approach is more performant in general eg. when the rowkey is a full rowkey as in our case. 

In summary, seeing as the HBase client API uses a Scanner under the hood when doing a Get, there should be no real benefit to having a Get added to the code (at least not without doing some practical benchmarks). 

I'll use the Scan for my processor instead since it already has the functionality I need. Will add the processor as a separate PR.

Can I close this PR, or does that need to be done on your side?

Also, I see that the automated checks have failed (looks like other components' tests). Is this something I should worry about for my next PR? :)

**Comment by wangrenlei** [comment]
@baolsen  you should check your style of code .

**Comment by bbende** [comment]
@baolsen Thanks for the info. I believe only you can close the pull request on your end.

As far as the automated checks, you can ignore appveryor, but the travis build should be passing as of yesterday. Looking at the travis logs for your build it shows:

[INFO] --- maven-checkstyle-plugin:2.15:check (check-style) @ nifi-hbase-client-service-api ---
[WARNING] src/main/java/org/apache/nifi/hbase/HBaseClientService.java[131] (regexp) RegexpSinglelineJava: Line has trailing whitespace.

You can detect this locally by running "mvn clean install -Pcontrib-check" before submitting anything. The contrib-check will fail if anything doesn't pass checkstyles or the RAT licensing check.

**Comment by baolsen** [comment]
Great, thanks for all the info. 
Will close this PR meanwhile.
